src/pybind/mgr/cephadm/serve.py

```diff
@@ -944,11 +944,16 @@ def _check_daemons(self) -> None:
                     dd.name()))
                 action = 'reconfig'
             elif last_deps != deps:
-                self.log.debug('%s deps %s -> %s' % (dd.name(), last_deps,
-                                                     deps))
-                self.log.info('Reconfiguring %s (dependencies changed)...' % (
-                    dd.name()))
+                self.log.debug(f'{dd.name()} deps {last_deps} -> {deps}')
+                self.log.info(f'Reconfiguring {dd.name()} (dependencies changed)...')
                 action = 'reconfig'
+                service_name = dd.name().split('.')[0]
+                # we need only redeploy if secure_monitoring_stack value has changed:
+                if service_name in ['prometheus', 'node-exporter', 'alertmanager']:
+                    diff = list(set(last_deps) - set(deps))
```

Comment:
using diff directly as "set" instead "list" make things more easy:

```
diff = set(last_deps) - set(deps)
if  'secure_monitoring_stack' in diff : action='redeploy' 
```

Reply:
> 

In this case in fact the dep is not `secure_monitoring_stack` but is `secure_monitoring_stack:[true|false]` depending on its bool value. So we can't just check if the value is in the set directly instead we need the list so we can iterate over all the items. I'll add a comment explaining this since it's not straightforward and somebody reading the code could get easily confused.